k8srm-prototype/pkg/api/capacity_types.go

```diff
@@ -53,6 +61,58 @@ type DevicePoolStatus struct {
 	AvailableDevices int `json:"availableDevices,omitempty"`
 }
 
+// Device is used to track individual devices in a pool.
+type Device struct {
+	// Name is a driver-specific identifier for the device.
+	// +required
+	Name string `json:"name"`
+
+	// Attributes contain additional metadata that can be used in
+	// constraints. If an attribute name overlaps with the pool attribute,
+	// the device attribute takes precedence.
+	//
+	// +optional
+	Attributes []Attribute `json:"attributes,omitempty"`
+
+	// Requests contains the pooled resources that are consumed when
+	// this device is allocated.
+	//
+	// +optional
+	Requests map[string]resource.Quantity `json:"requests,omitempty"`
+
+	// Resources allows the definition of per-device resources that can
+	// be allocated in a manner similar to standard Kubernetes resources.
+	// NOTE: We may not need to implement this right away.
+	//
+	// +optional
+	Resources []ResourceCapacity `json:"resources,omitempty"`
```

Comment:
Tim: this should probably be called "allocatable" - or whatever node currently uses for this.